syntax = "proto3"; 

import "battery_status.proto"; 

package bos; 
message GetStatus {

}
message ScheduleSetCurrent {
    sint64 target_current_mA = 1; 
    CTimeStamp when_to_set = 2; 
    CTimeStamp until_when = 3; 
}
message GetInfo {

}
message BatteryMsg {
    oneof func_call {
        GetStatus get_status = 1; 
        ScheduleSetCurrent schedule_set_current = 2; 
        GetInfo get_info = 3; 
    }
}



